src/main/java/uk/gov/companieshouse/api/accounts/service/impl/CompanyAccountServiceImpl.java

```diff
@@ -2,47 +2,36 @@
 
 import java.util.HashMap;
 import java.util.Map;
-import javax.annotation.Resource;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.data.mongodb.repository.MongoRepository;
 import org.springframework.stereotype.Service;
-import uk.gov.companieshouse.GenerateEtagUtil;
 import uk.gov.companieshouse.api.accounts.Kind;
 import uk.gov.companieshouse.api.accounts.LinkType;
 import uk.gov.companieshouse.api.accounts.model.entity.CompanyAccountEntity;
 import uk.gov.companieshouse.api.accounts.model.rest.CompanyAccount;
-import uk.gov.companieshouse.api.accounts.repository.CompanyAccountRepository;
 import uk.gov.companieshouse.api.accounts.service.CompanyAccountService;
-import uk.gov.companieshouse.api.accounts.transformer.CompanyAccountTransformer;
+import uk.gov.companieshouse.api.accounts.transformer.GenericTransformer;
 
 @Service
-public class CompanyAccountServiceImpl implements CompanyAccountService {
+public class CompanyAccountServiceImpl extends AbstractServiceImpl<CompanyAccount, CompanyAccountEntity> implements CompanyAccountService  {
 
     @Autowired
-    private CompanyAccountRepository companyAccountRepository;
-
-    @Resource
-    private CompanyAccountTransformer companyAccountTransformer;
-
-    /**
-     * {@inheritDoc}
-     */
-    public CompanyAccount createCompanyAccount(CompanyAccount companyAccount) {
-        generateEtagLinksKind(companyAccount);
-
-        CompanyAccountEntity companyAccountEntity = companyAccountTransformer.transform(companyAccount);
-
-        companyAccountRepository.insert(companyAccountEntity);
-
-        return companyAccount;
+    public CompanyAccountServiceImpl(
+            @Qualifier("companyAccountRepository") MongoRepository<CompanyAccountEntity, String> mongoRepository,
+            @Qualifier("companyAccountTransformer") GenericTransformer<CompanyAccount, CompanyAccountEntity> companyAccountTransformer) {
+        super(mongoRepository, companyAccountTransformer);
     }
 
-    private void generateEtagLinksKind(CompanyAccount companyAccount) {
-        companyAccount.setEtag(GenerateEtagUtil.generateEtag());
-        companyAccount.setKind(Kind.ACCOUNT.getValue());
-
-        Map<String, String> links = new HashMap<>();
+    @Override
+    public void addLinks(CompanyAccount rest) {
+//        Map<String, String> links = new HashMap<>();
+//        links.put(LinkType.SELF.getLink(), "");
+//        rest.setLinks(links);
```

Comment:
Can you remove the commented out code or add a TODO linked to a story related to it please?

Commented out code is bad practise because it confuses devs in the future whether it was meant to be added, removed or doesn't work. See: https://companieshouse.atlassian.net/wiki/spaces/DEV/pages/142213166/Readable+Maintainable+Testable+Elegant+Code#Readable,Maintainable,Testable,ElegantCode-Comments